internal/cyclonedx/cyclonedx.go

```diff
@@ -14,18 +14,22 @@ import (
 	"github.com/tidelift/tidelift-sbom-info/internal/utils"
 )
 
-func SupportedPurlsFromBomFile(filename string) ([]packageurl.PackageURL, error) {
+func SupportedPurlsFromBomFile(filename string) (string, []packageurl.PackageURL, error) {
 	bom, err := decodeCyclonedx(filename)
 	if err != nil {
-		return nil, err
+		return "", nil, err
 	}
 
 	purls, err := extractSupportedPurls(bom)
 	if err != nil {
-		return nil, err
+		return "", nil, err
 	}
 
-	return purls, nil
+	if bom.Metadata == nil || bom.Metadata.Component == nil {
+		log.Warn("CycloneDX file does not have any metadata")
+		return "", purls, nil
+	}
+	return bom.Metadata.Component.Name, purls, nil
```

Comment:
This theoretically lets us have a project name that matches whatever the toplevel item in the SBOM is. Our SBOMs don't populate this, though.

Reply:
should we just put this in a different method, e.g. `ProjectNameFromBomFile()`, or rename the method, e.g. `NameAndPurlsFromBomFile()`?

Reply:
as this is the only function that actually touches the BOM object, decided to put it here rather than refactor it